File: api/resources/reprocessing.go
package resources

import (
	"github.com/ONSBR/Plataforma-UI/api/services"
	"github.com/labstack/echo"
)

func FindallReprocessing(c echo.Context) error {
	service := services.NewReprocessingService()

	instances, err := service.Findall(c.QueryParam("systemId"), c.QueryParam("status"))
	if err != nil {
		return err
	}
	return c.JSON(200, instances)
}

func ApproveReprocessing(c echo.Context) error {
	service := services.NewReprocessingService()

	instances, err := service.Approve(c.QueryParam("reprocessingId"), c.QueryParam("approver"))
	if err != nil {
		return err
	}
	return c.JSON(200, instances)
}

func SkipReprocessing(c echo.Context) error {
	service := services.NewReprocessingService()

	instances, err := service.Skip(c.QueryParam("reprocessingId"), c.QueryParam("approver"))
	if err != nil {
		return err
	}
	return c.JSON(200, instances)
}

File: frontend/ui/src/Services/api/reprocessing.js
import ApiService from './index'
import Axios from 'axios'

class ReprocessingService extends ApiService {
    constructor(){
        super()
    }

    findAll(systemId, status){
        if (status){
            return Axios.get(this.url("reprocessing/findall",{ systemId, status}))
        }else{
            return Axios.get(this.url("reprocessing/findall",{systemId}))
        }
    }

    approve(reprocessingId, approver){
        return Axios.post(this.url("reprocessing/approve",{reprocessingId,approver}))
    }

    skip(reprocessingId, approver){
        return Axios.post(this.url("reprocessing/skip",{reprocessingId,approver}))
    }

    override(systemId, reprocessingId, status) {
        return Axios.post(this.url("reprocessing/overrideStatus",{systemId,reprocessingId,status}))
    }

}

export default ReprocessingService